src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -650,6 +650,59 @@ private TypeElement getElement(Class<?> clazz) {
         + "}\n");
   }
 
+  @Test public void annotation() throws Exception {
+    TypeSpec annotation = TypeSpec.annotationBuilder("MyAnnotation")
+        .addModifiers(Modifier.PUBLIC)
+        .addMethod(MethodSpec.methodBuilder("test")
+                .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+                .defaultValue("$L", 0)
+                .returns(int.class)
+                .build())
+        .build();
+
+    assertThat(toString(annotation)).isEqualTo(""
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "public @interface MyAnnotation {\n"
+        + "  int test() default 0;\n"
+        + "}\n"
+    );
+  }
+
+  @Test public void innerAnnotationInAnnotationDeclaration() throws Exception {
+    TypeSpec bar = TypeSpec.annotationBuilder("Bar")
+        .addMethod(MethodSpec.methodBuilder("value")
+            .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+            .defaultValue("@$T", Deprecated.class)
+            .returns(Deprecated.class)
+            .build())
+        .build();
+
+    assertThat(toString(bar)).isEqualTo(""
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "import java.lang.Deprecated;\n"
+        + "\n"
+        + "@interface Bar {\n"
+        + "  Deprecated value() default @Deprecated;\n"
```

Comment:
Note: we should use `AnnotationSpec` in this test; `("@$T", Deprecated.class)` is a hack. Now maybe using the `toString` of `AnnotationSpec` would be enough (I must say it's already too long since I looked closely at JavaPoet's code)